linter/lints/cpcps/lint_crl_has_idp.go

```diff
@@ -142,41 +140,35 @@ func (l *crlHasIDP) Execute(c *x509.RevocationList) *lint.LintResult {
 				Details: "IDP should contain only one distributionPoint",
 			}
 		}
-		if !idpv.ReadASN1Bytes(&onlyContainsUserCerts, cryptobyte_asn1.Tag(1).ContextSpecific()) {
+
+		var onlyContainsUserCertsBytes []byte
+		if !idpv.ReadASN1Bytes(&onlyContainsUserCertsBytes, onlyContainsUserCertsTag) {
 			return &lint.LintResult{
 				Status:  lint.Error,
 				Details: "Failed to read IDP onlyContainsUserCerts",
 			}
 		}
-		if len(onlyContainsUserCerts) != 1 || onlyContainsUserCerts[0] != 0xFF {
+		if !lints.ReadASN1BooleanWithTag(onlyContainsUserCertsBytes) {
 			return &lint.LintResult{
 				Status:  lint.Error,
 				Details: "IDP should set onlyContainsUserCerts: TRUE",
 			}
 		}
 	} else {
-		// We read this boolean as a byte and ensure the value is 0xFF because
-		// cryptobyte.ReadASN1Boolean can't handle the custom encoding rules for
-		// the onlyContainsUserCerts[1] and onlyContainsCACerts[2] tagged
-		// fields. X.690 (07/2002) section 8.2 states that a boolean set to true
-		// will have contents FF.
-		// https://www.itu.int/rec/T-REC-X.690-200207-S/en
-		onlyContainsCACerts := make([]byte, 0)
-
-		if idpv.PeekASN1Tag(cryptobyte_asn1.Tag(2).ContextSpecific()) {
-			// If either the onlyContainsCACerts[2] field is present, assume we're
-			// dealing with a CRL containing CA Certs.
-			if !idpv.ReadASN1Bytes(&onlyContainsCACerts, cryptobyte_asn1.Tag(2).ContextSpecific()) {
-				return &lint.LintResult{
-					Status:  lint.Error,
-					Details: "Failed to read IDP onlyContainsCACerts",
-				}
+		// If the onlyContainsCACerts [2] field is present, assume we're dealing
+		// with a CRL containing CA Certs.
```

Comment:
Removing the redundant peek here is great, but now this comment doesn't precisely match the logic as implemented. Instead we should say: If neither the distributionPoint nor the onlyContainsUserCerts field was present, assume we're dealing with a CRL containing CA certs, and require the onlyContainsCACerts field.